File: tarea2.java
package tarea2_121;

import java.awt.*;
import javax.swing.*;

class Punto {
    int x, y;

    public Punto(int x, int y) {
        this.x = x;
        this.y = y;
    }

    public int[] coordCartesianas() {
        return new int[]{x, y};
    }

    public double[] coordPolares() {
        double radio = Math.sqrt(x * x + y * y);
        double angulo = Math.atan2(y, x);
        return new double[]{radio, angulo};
    }

    @Override
    public String toString() {
        return "(" + x + ", " + y + ")";
    }
}

class Linea {
    Punto p1, p2;

    public Linea(Punto p1, Punto p2) {
        this.p1 = p1;
        this.p2 = p2;
    }

    public void dibujarLinea(Graphics g) {
        g.drawLine(p1.x * 20, p1.y * 20, p2.x * 20, p2.y * 20);
    }

    @Override
    public String toString() {
        return "Punto 1: " + p1 + ", Punto 2: " + p2;
    }
}

class Circulo {
    Punto centro;
    int radio;

    public Circulo(Punto centro, int radio) {
        this.centro = centro;
        this.radio = radio;
    }

    public void dibujarCirculo(Graphics g) {
        g.drawOval(centro.x * 20 - radio * 20, centro.y * 20 - radio * 20, radio * 40, radio * 40);
    }

    @Override
    public String toString() {
        return "Centro: " + centro + ", Radio: " + radio;
    }
}

public class tarea2 extends JPanel {
    Punto punto1 = new Punto(2, 3);
    Punto punto2 = new Punto(4, 5);
    Linea linea = new Linea(punto1, punto2);
    Circulo circulo = new Circulo(punto1, 2);

    @Override
    protected void paintComponent(Graphics g) {
        super.paintComponent(g);
        circulo.dibujarCirculo(g);
        linea.dibujarLinea(g);
    }

    public static void main(String[] args) {
        Punto punto1 = new Punto(2, 3);
        Punto punto2 = new Punto(4, 5);
        Linea linea = new Linea(punto1, punto2);
        Circulo circulo = new Circulo(punto1, 2);

        int[] cartesianas = punto1.coordCartesianas();
        double[] polares = punto1.coordPolares();

        System.out.println("Coordenadas cartesianas:");
        System.out.println(cartesianas[0] + " " + cartesianas[1]);

        System.out.println("Coordenadas polares:");
        System.out.println(polares[0] + " " + polares[1]);

        System.out.println(punto1 + "  Punto 1");
        System.out.println(punto2 + "  Punto 2");
        System.out.println(linea + "  Línea");

        JFrame frame = new JFrame();
        tarea2 dibujo = new tarea2();
        frame.add(dibujo);
        frame.setSize(400, 400);
        frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
        frame.setVisible(true);
    }
}

File: tarea2.py
# segundo  ejercicio
import math
import turtle
class Punto:
    def __init__(self,x,y):
        self.x=x
        self.y=y
    def coord_cartesianas(self):
        return self.x, self.y
    def coord_polares(self):
        radio=math.sqrt(self.x*self.x+self.y*self.y)
        angulo=math.atan(self.x/self.y)
        return radio, angulo
    def __str__(self):
        return str(self.x)+" " +str(self.y)
class Linea:
    def __init__(self,p1,p2):
        self.p1=p1
        self.p2=p2

    def dibujalinea(self,linea):
        linea.penup()
        linea.goto(self.p1.x*20, self.p1.y*20)
        linea.pendown()
        linea.goto(self.p2.x*20, self.p2.y*20)
        turtle.done()
    def __str__(self):
        return str(f"x: ({self.p1})")+" " +str(f"y: ({self.p2})")
class Circulo:
    def __init__(self,puntocirculo,radio):
        self.puntocirculo=puntocirculo
        self.radio=radio
    def dibujacirculo(self,circulo):
        circulo.penup()
        circulo.goto(self.puntocirculo.x*20,self.puntocirculo.y*20 - self.radio*20)
        circulo.pendown()
        circulo.circle(self.radio*20)
    def __str__(self):
        return str(f" puntos del circulo {self.puntocirculo}")+str(f"el radio del circulo {self.radio}")
punto1=Punto(2,3)
punto2=Punto(4,5)
linea=Linea(punto1,punto2)
circulo=Circulo(punto1,2)
x,y=punto1.coord_cartesianas()
r,a=punto1.coord_polares()

print("coordenadas cartesianas")
print(x,y)
print("coordenadas polares")
print(r,a)

print(punto1, "  punto 1")
print(punto2, "  punto 2")
print(linea,"  linea")

screen=turtle.Screen()
circulo_linea=turtle.Turtle()

circulo.dibujacirculo(circulo_linea)
linea.dibujalinea(circulo_linea)
turtle.done()